assets/queries/terraform/aws/port22_in_public_scope/query.rego

```diff
@@ -14,23 +14,37 @@ isPrivate(cidr) = allow {
 	count({x | cidr[x]; cidr[x] == privateIPs[j]}) == cidrLength
 
 	allow = true
+} else = blocksInVars(cidr)
+
+blocksInVars(cidr) { # check if all CIDR blocks are stored in variables, which cannot be evaluated in the static scanning
+    len := count(cidr)
+    count({b | cidr[b]; contains(cidr[b],"$")}) == len
```

Comment:
I think the query will return a result if the ingress has a list of cidr_blocks with private IP's and variables (e.g. `["192.120.0.0/16", "75.132.0.0/16", "${var}"]`).

A better approach could be checking if some CIDR block is neither a private IP nor a variable.

Reply:
Attended request